docs(idempotency): fix misleading idempotent examples (#661)
awslabs/aws-lambda-powertools-python: docs/utilities/idempotency.md

Imagine the function executes successfully, but the client never receives the response due to a connection issue. It is safe to retry in this instance, as the idempotent decorator will return a previously saved response.

> !!! warning "Making JSON APIs Idempotent"

Reviewer: Making this a little more generic as it applies outside of API use cases
```suggestion
!!! warning "Idempotency for JSON payloads"
```

>     To ensure JSON structured APIs are fully idempotent we must ensure the idempotent key is a valid JSON object.

Reviewer: ```suggestion
    The payload extracted by the `event_key_jmespath` is treated as a string by default, so will be sensitive to differences in whitespace even when the JSON payload itself is identical.
```

>     To do this we can [customize the default behaviour](#customizing-the-default-behavior) with *event_key_jmespath* and the [JMESPath built-in function](/utilities/jmespath_functions) *powertools_json()*.

Reviewer: ```suggestion
    To alter this behaviour, we can use the [JMESPath built-in function](/utilities/jmespath_functions) *powertools_json()* to treat the payload as a JSON object rather than a string.
```
